sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -283,6 +290,13 @@ abstract class ProgressContext(
     progressReporter.lastNoExecutionProgressEventTime = triggerClock.getTimeMillis()
     progressReporter.updateProgress(newProgress)
 
+    // Ask the state store coordinator to log all lagging state stores
+    if (progressReporter.coordinatorReportSnapshotUploadLag) {
```

Comment:
nit: should this check to see if it is enabled be done in the `logLaggingStateStores` func?

Reply:
We do it in both sides right now - this check is added here for both readability and reducing overhead to retrieve the coordinator ref + sending the RPC call